id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java

```diff
@@ -340,4 +359,20 @@ public void updateCredentialStatus(CredentialStatusEvent credentialStatusEvent)
 		
 	}
 	
+	public void audit(AuditModules module, AuditEvents event, String id, IdType idType, String desc) {
+		RequestWrapper<AuditRequestDTO> auditRequest = auditBuilder.buildRequest(module, event,
+				securityManager.hash(id.getBytes()), idType, desc);
+		HttpEntity<RequestWrapper<AuditRequestDTO>> httpEntity = new HttpEntity<>(auditRequest);
+		try {
+			restUtil.postApi(ApiName.KERNELAUDITMANAGER, null, id, desc, MediaType.APPLICATION_JSON, httpEntity,
+					AuditResponseDTO.class);
+		} catch (IdRepoDataValidationException e) {
+			LOGGER.error(IdRepoSecurityManager.getUser(), "AuditRequestBuilder", "audit",
+					"Exception : " + ExceptionUtils.getStackTrace(e));
+		} catch (Exception e) {
+			LOGGER.error(IdRepoSecurityManager.getUser(), "AuditRequestBuilder", "audit",
```

Comment:
I think for this and above, "AuditRequestBuilder" should be "CredentialRequestServiceImpl"

Reply:
Resolved